### ISSUE TITLE
D13: `evaluate` and `inspect_failures` are gated behind the weights container

### ISSUE THREAD

**Issue by ash-ding** [ash-ding/Glyph#13]
`tool_defs` puts `evaluate` and `inspect_failures` inside
`if Container.WEIGHTS in allowed:`, so the arms do not have comparable
self-evaluation:

| | A2 | A4 | A6 |
|---|---|---|---|
| tools | 3 | 3 | 7 |
| self-evaluation | **none** | `write_code(check_on=...)` | evaluate / inspect / retrain |
| what it did (rerun) | query x5, set_context, seal | -- | query x11, synth x3, train x3, evaluate x2, inspect x2, seal |

A2 writes its prompt blind and seals.

This matters because the previous entry established that the dev signal
strongly shapes behaviour: once dev stopped reading zero, the agent went from
187 purchased facts to 565. So **A2's 0.055 and A6's 0.035 were not produced
under equal feedback** -- and by #9 they were not distinguishable anyway.

Whether an arm can score itself is a harness choice, not a property of the
container, so as it stands it is a confound in the comparison.

Options: (a) move both out of the gate so all three arms can evaluate their own
artifact on purchased dev items; (b) remove A4's `check_on` too so all three
write blind -- but iteration is a genuine advantage of the weights route, so
this removes signal; (c) leave it and report the asymmetry.

**Recommendation: (a).** Self-evaluation is available to every route in
practice; it should not be issued to one arm only.

**Comment by ash-ding** [comment]
**Settled: self-assessment is universal**, shipped in `ebbe725`.

It is not a property of a container — someone writing a prompt can try it,
someone writing code can run it, someone training a model can hold out a
validation set. Issuing it to one arm measured the harness rather than the
container.

**Ungating was not enough, and that is the part worth recording.** `evaluate`
was checkpoint-only by construction:

```python
path = self.checkpoints.get(checkpoint_id)
if not path: return {"error": ...}
```

A2 holds a string and A4 holds source code, so neither had an id to pass —
moving the tool out of the container block would have handed them a tool that
returns an error. It was written to evaluate *an adapter*, not *what I made*.

The capability already existed one layer up: `seal.evaluate()` takes an
`answer_fn` and each arm runner supplied one. That is now factored into
`seal.answer_with`, used by the tool layer at dev time and by all three arm
runners at test time.

The second benefit outweighs the first. Two implementations could drift —
different answer cleaning, different caching — and the agent would then be
steering on a number produced differently from the one it is finally graded by,
which is invisible from either side. One path makes that impossible rather than
unlikely, and #14's calibration measurement depends on it.

Also landed with it: declaration and allocation are now separate tables in
`schema.py`, `set_context`/`write_code`/`train` return artifact ids and `seal`
takes one, `train` is always a full fine-tune, and A7 became a row rather than a
module. See `docs/tools.md`.
